src/lib/socket.c

```diff
@@ -57,6 +57,42 @@
 #include <openssl/ssl.h>
 #include <openssl/err.h>
 #include <openssl/bio.h>
+#if OPENSSL_VERSION_NUMBER < 0x10100001L
```

Comment:
I think their answer is the same as for musl libc: build-time feature detection a la autoconf.